_posts/2017/2017-07-20-fullstack-london-of-dragons-and-numbers.md

```diff
@@ -0,0 +1,124 @@
+Decorators are an excellent way to let all of your functions be responsible for a single purpose and "inject" every other (reusable) behaviour, such as logging or measuring.
+This insight does not only apply to TypeScript but can also be used when decorators come (back) to the JavaScript language since they are basically sugar for higher-order functions.
+
+The most interesting example was a timeout decorator that could be used to let an async operation fail if it took longer than specified.
+Therefore the decorator let the original async operation race against a delayed rejecting promise.
+If the operation completed before the rejecting promise it was considered successful.
+As soon as it took longer, the delayed promise would reject and fail the whole operation marking it as timed out.
+
+## npm vs gulp vs webpack vs what the heck!?
+
+Although we're using all of these tools in daily development and this was labeled a beginner session, [Rubén Sospedra](https://twitter.com/sospedra_r) made you rethink the usage scenarios of the above mentioned.
+
+It was no _X_ is better than _Y_ is better than _Z_ talk which I was very grateful for.
+Every tool has its right to exist and should be used accordingly.
+Use npm for specifying simple commands or commands that trigger a chain of commands (a pipeline).
+Then specify complex pipelines with gulp (or any task runner that is hip at the time of reading).
+Finally let webpack (or rollup or one of [substack's](https://github.com/substack) epic projects) take care of bundling your application.
+
+## Blockchain 101
+
+Since the rise of Bitcoin everybody heard about crypto currencies and a mysterious construct that is the blockchain.
+IBM's [Kevin Hoyt](https://twitter.com/krhoyt) did a great job demystifying this "distributed database with special properties".
+And that's what it is (basically).
+When you think about it, the whole Internet could be built on blockchain technology 🤔 😱
+
+Watch the [skills cast](https://skillsmatter.com/skillscasts/10360-understanding-blockchain), it'll open your eyes.
+
+## Benchmarking and performance
+
+There were two talks I attended that dealt with performance and how to measure it.
+
+> Sometimes measuring performance is art rather than science.
+
+<!-- change the next sentence somehow -->
+The above quote's author [Ahmad Nassri](https://twitter.com/AhmadNassri) laid out a pretty relevant case of measuring function call performance especially in the time of AWS, Azure and GCE.
+The faster your function executes in these environments the lesser you pay.
+It's that easy.
+Not only should benchmarking be part of the after-release phase (which is basically monitoring at that point) but also during the development cycle.
+However, do not overdo optimizing (especially not premature).
+
+[Martin Splitt](https://twitter.com/g33konaut) on the other hand was shedding some light on browser rendering performance.
```

Comment:
[Martin Splitt](https://twitter.com/g33konaut), on the other hand, shed some light on...